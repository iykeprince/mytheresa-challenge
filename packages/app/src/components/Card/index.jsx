import React from "react";

import "@mytheresa/scss/lib/Card.css";

const Card = ({ title, url }) => {
  return (
    <div data-testid="MytCard" className="myt-card__container">
      <img
        src={`https://image.tmdb.org/t/p/original${url}`}
        alt={title}
        className="myt-card__image"
      />
      <div className="myt-card__caption">{title}</div>
    </div>
  );
};

export default Card;
