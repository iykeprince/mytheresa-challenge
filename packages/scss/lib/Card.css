/*=============================================
    #Variables
=============================================*/
/**
 * TABLE OF CONTENTS
 * 
 * COLORS
 * Brand colors...................These are the main brand colors. They will be used the most
 * Neutral colors.................White and black, gray scale, secondary colors.
 * Utility colors.................Other colors to represent states: info, warning, success, etc.
 * 
 * TYPOGRAPHY
 * Font families..................The fonts used in the design system.
 * Font weights...................Font weights
 * Sizes..........................Font sizing for body text, paragraphs, inputs
 * Heading sizes..................Font sizing for page headings
 *
 * BREAKPOINTS
 * Breakpoints....................Global responsive breakpoints definitions.
 *
 * SPACING
 * Spacing........................Global fixed set of spacing variables
 */
/*=============================================
=                  Colors                     =
=============================================*/
/* Brand Colors */
/* Neutral Colors */
/* Utility colors */
/*=====  End of Colors  ======*/
/*=============================================
=                Typography                   =
=============================================*/
/**
 *
 * 1) The playfair font. We'll use this for headings mostly.
 *     https://fonts.google.com/specimen/Playfair+Display
 *
 * 2) The raleway font. We'll use this for the body text.
 *     https://fonts.google.com/specimen/Raleway
 */
/* Font families */
/* 1 */
/* 2 */
/* Font sizes */
/* Heading font sizes */
/* Font weights */
/*=====  End of Typography  ======*/
/*=============================================
=            Breakpoints                      =
=============================================*/
/*=====  End of Breakpoints  ======*/
/*=============================================
=            Spacing                          =
=============================================*/
/*=====  End of Spacing  ======*/
/*=============================================
=            Foundation - colors              =
=============================================*/
/**
 * This file defines the actual colors that will be used for styling. They will default to the palette
 * we defined in the _variable.scss file. This is our default palette, and devs can override this
 * with their own variables.
 */
/*=============================================
=            Global text colors               =
=============================================*/
/*=============================================
=                 Buttons                     =
=============================================*/
/*=============================================
=                 Forms                       =
=============================================*/
/*=============================================
=        Foundation - Typography              =
=============================================*/
/**
 * This file defines the default font sizes to be used on headings and texts. The defaults come
 * from the palette defined in _variables.scss.
 */
/*=============================================
=            Body font family                  =
=============================================*/
/*=============================================
=            Body font sizes                  =
=============================================*/
/*=============================================
=            Heading font sizes               =
=============================================*/
/*=============================================
=            Head font weights                =
=============================================*/
/*=============================================
=            Mixins                           =
=============================================*/
/**
 *
 * Helpful reusable function-like things that will be used to implement
 * styles for components.
 *
 */
.myt-card__container {
  box-shadow: 0 4px 11px rgba(157, 157, 157, 0.21);
  border-radius: 8px;
  background-color: #fff;
  width: 20rem;
  height: 20rem;
  margin-right: 1rem;
  position: relative;
  cursor: pointer;
}

.myt-card__container .myt-card__image {
  width: 100%;
  height: 100%;
  border-radius: 8px;
}

.myt-card__container .myt-card__caption {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0, 0, 0, 0.6);
  padding: 1rem;
  font-size: var(--myt-h4-font-size, 1.125rem);
  color: #fff;
  font-family: "Raleway", "HelveticaNeue", "Helvetica", sans-serif;
  line-height: 1.2rem;
}
