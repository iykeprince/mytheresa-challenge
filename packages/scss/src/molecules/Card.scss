@import "foundation/all";

.myt-card__container {
  box-shadow: map-get($shadows, 'base');
  border-radius: 8px;
  background-color: $white;
  width: 20rem;
  height: 20rem;
  margin-right: 1rem;
  position: relative;
  cursor: pointer;


  .myt-card__image {
    width: 100%;
    height: 100%;
    border-radius: 8px;
    ;
  }

  .myt-card__caption {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: rgba($color: #000000, $alpha: 0.6);

    padding: map-get($spacing, 'sm');
    font-size: $h-4-font-size;
    color: $white;
    font-family: $raleway;
    line-height: 1.2rem;

  }

}