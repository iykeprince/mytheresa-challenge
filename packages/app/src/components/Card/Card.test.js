import React from 'react'
import Card from './index'
import { render } from '@testing-library/react'

test('should render card', () => {
  const { getByTestId } = render(<Card />)

  expect(getByTestId('MytCard')).toBeDefined()
})

test('snapshot of the card', () => {
  const { asFragment } = render(<Card />)

  expect(asFragment()).toMatchSnapshot()
})

test('should render test prop passed to it', () => { })