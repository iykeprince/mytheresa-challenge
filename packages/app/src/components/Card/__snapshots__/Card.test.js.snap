// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the card 1`] = `
<DocumentFragment>
  <div
    class="myt-card__container"
    data-testid="MytCard"
  >
    <img
      class="myt-card__image"
      src="https://image.tmdb.org/t/p/originalundefined"
    />
    <div
      class="myt-card__caption"
    />
  </div>
</DocumentFragment>
`;
